lib/manager/pre-commit/extract.ts

```diff
@@ -0,0 +1,178 @@
+import is from '@sindresorhus/is';
+import yaml from 'js-yaml';
+import {
+  PLATFORM_TYPE_GITEA,
+  PLATFORM_TYPE_GITHUB,
+  PLATFORM_TYPE_GITLAB,
+} from '../../constants/platforms';
+import { id as githubTagsId } from '../../datasource/github-tags';
+import { id as gitlabTagsId } from '../../datasource/gitlab-tags';
+import { logger } from '../../logger';
+import { SkipReason } from '../../types';
+import { find } from '../../util/host-rules';
+import { regEx } from '../../util/regex';
+import { PackageDependency, PackageFile } from '../common';
+
+import {
+  matchesPrecommitConfigHeuristic,
+  matchesPrecommitDependencyHeuristic,
+} from './parsing';
+import { PreCommitConfig } from './types';
+
+function isEmptyObject(obj: any): boolean {
+  return Object.keys(obj).length === 0 && obj.constructor === Object;
+}
+
+/**
+ * Determines the datasource(id) to be used for this dependency
+ * @param repository the full git url, ie git@github.com/user/project.
+ *        Used in debug statements to clearly indicate the related dependency.
+ * @param hostName the hostname (ie github.com)
+ *        Used to determine which renovate datasource should be used.
+ *        Is matched literally against `github.com` and `gitlab.com`.
+ *        If that doesn't match, `hostRules.find()` is used to find related sources.
+ *        In that case, the hostname is passed on as registryUrl to the corresponding datasource.
+ */
+function determineDatasource(
+  repository: string,
+  hostName: string
+): { datasource?: string; registryUrls?: string[]; skipReason?: SkipReason } {
+  if (hostName === 'github.com') {
+    logger.debug({ repository, hostName }, 'Found github dependency');
+    return { datasource: githubTagsId };
+  }
+  if (hostName === 'gitlab.com') {
+    logger.debug({ repository, hostName }, 'Found gitlab dependency');
+    return { datasource: gitlabTagsId };
+  }
+  const res = find({ url: hostName });
```

Comment:
This will only work if the `hostName` is a valid URL, e.g. starting with `https://`

Reply:
'k, let me then prefix the hostname with "https://", as the hostname can also have been extracted from a git@ url (thus can't take the original).
Is there already a stub `hostRules` used somewhere, such that I can actually test the interaction? This code is touched by the tests, but issues like the missing `https` aren't truly tested currently.